src/Servers/HttpSys/src/RequestProcessing/Response.cs

```diff
@@ -402,12 +405,25 @@ internal HttpApiTypes.HTTP_FLAGS ComputeHeaders(long writeCount, bool endOfReque
 
         // Determine if the connection will be kept alive or closed.
         var keepConnectionAlive = true;
-        if (requestVersion <= Constants.V1_0 // Http.Sys does not support "Keep-Alive: true" or "Connection: Keep-Alive"
+
+        if (requestVersion < Constants.V1_0
             || (requestVersion == Constants.V1_1 && requestCloseSet)
             || responseCloseSet)
         {
             keepConnectionAlive = false;
         }
+        else if (requestVersion == Constants.V1_0)
+        {
+            // In .NET 9, we updated the behavior for 1.0 clients here to match
+            // RFC 2068. The new behavior is available down-level behind an
+            // AppContext switch.
+
+            // An HTTP/1.1 server may also establish persistent connections with
+            // HTTP/1.0 clients upon receipt of a Keep-Alive connection token.
+            // However, a persistent connection with an HTTP/1.0 client cannot make
+            // use of the chunked transfer-coding. From: https://www.rfc-editor.org/rfc/rfc2068#section-19.7.1
+            keepConnectionAlive = _respectHttp10KeepAlive && requestConnectionKeepAliveSet && !responseChunkedSet;
```

Comment:
responseChunkedSet is only for cases where the application sets the chunked header and does the chunking themselves, it doesn't cover the default case where the server does chunking automatically. See `_boundaryType = BoundaryType.Chunked;` vs `_boundaryType = BoundaryType.PassThrough;` below.

Reply:
Thanks for the review!